### PR TITLE
Fix logging of MID matching.

### PR DESCRIPTION
**Pull request by boaks**
Recreated request from observation stores may not have a valid MID.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

**Comment by boaks** [comment]
Intended for issue #668

File: californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java [hunk 1]
```diff
@@ -112,6 +112,7 @@ public void sendRequest(final Exchange exchange) {
 		// for observe request.
 		Request request = exchange.getCurrentRequest();
 		if (request.isObserve() && 0 == exchange.getFailedTransmissionCount()) {
+			exchangeStore.assignMessageId(request);
```
Comment (sbernard31) [inline californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java]: Should we handle the case where `assignMessageId` returns `Message.NONE` ?

**Comment by boaks** [comment]
Check for exhausted MID added.

**Comment by sbernard31** [comment]
:+1:

**Comment by boaks** [comment]
So in some time in the future, we hopefully will see, what caused the warning of #668. :-)

**Pull request merged**